src/profiler.cpp

```diff
@@ -1075,6 +1088,30 @@ Error Profiler::start(Arguments& args, bool reset) {
     return error;
 }
 
+Error Profiler::setEcid(const char* ecid) {
+    pthread_once(&local_ecid_key_once, createEcidKey);
+
+    void *value = pthread_getspecific(local_ecid_key);
+    destroyEcidKey(value);
+
+    int size = strlen(ecid);
+    char *duplicate = (char*) malloc( size+ 1 );
+    strcpy(duplicate, ecid);
+    int status = pthread_setspecific(local_ecid_key, duplicate);
+    return status == 0 ? Error::OK : Error("Cannot set pthread_setspecific");
+}
+
+const char* Profiler::getEcid() {
+    pthread_once(&local_ecid_key_once, createEcidKey);
```

Comment:
I'm not sure if this is async-signal-safe.

Reply:
Do you know if ```pthread_key_create``` should be invoked in every thread or just once in some static block? 

I found both implementation in the internet, and I'm not sure if this line is really needed.

Reply:
`pthread_key_create` needs to be called just once, and not inside a signal handler

Reply:
done